### PR TITLE
[PVR] Support for Kodi Eventlog

### PR DESCRIPTION
**Pull request by ksooo**
This PR extends Kodi eventlog (Jarvis feature) with some PVR events, namely
- create/delete timers
- start/stop recordings.

Ideas for other events are welcome.

@da-anda @Jalle19 mind taking a look

### CODE REVIEW

File: xbmc/pvr/addons/PVRClients.h [hunk 1]
```diff
@@ -155,12 +155,21 @@ namespace PVR
...
-     * @brief Get the friendly name for the client with the given id.
+     * @brief Get the (friendly) name for the client with the given id.
      * @param iClientId The id of the client.
-     * @param strName The friendly name of the client or an empty string when it wasn't found.
+     * @param strName The (friendly) name of the client or an empty string when it wasn't found.
+     * @param bFriendly return friendly vs. plain addon name
      * @return True if the client was found, false otherwise.
      */
-    bool GetClientName(int iClientId, std::string &strName) const;
+    bool GetClientName(int iClientId, std::string &strName, bool bFriendly = true) const;
```
Comment (AlwinEsch) [inline xbmc/pvr/addons/PVRClients.h]: For what is the ```bFriendly``` used? Some work in future?

Otherwise it looks good.

**Comment by FernetMenta** [comment]
if I am not mistaken, the decision was that only fixes make it to Jarvis. All other have to wait for 17.0
we intend to branch 17.0 soon

File: xbmc/pvr/addons/PVRClients.h (hunk 1 above)
Comment (ksooo) [inline xbmc/pvr/addons/PVRClients.h]: It's more the other way around. Original implemenation of CPVRClients::GetClientName() used always CPVRClient::GetFriendlyName(), but I needed just the plain addon name (CAddon::Name) for the eventlog entries. Thus, every existing call to CPVRClient::GetClientName actually uses "friendly name".

**Comment by ksooo** [comment]
@MartijnKaijser feel free to retarget thi PR, just thought that the new eventlog feature could need some love... At least my eventlog is almost always empty (except "Kodi sucessfully started") and therefore adds not too much value.

**Comment by MartijnKaijser** [comment]
iirc we would still allow minor non-API "feature" changes during beta

**Comment by FernetMenta** [comment]
I don't consider use of activity log as minor. This can have a big impact. It's not a self contained little feature

**Comment by ksooo** [comment]
Event log is in master for quite some time now and has proven to work. I do not see any risk in writing some more log entries. We're not talking about mass data or use of untested functionality here.

**Comment by da-anda** [comment]
some backends do really weird things with timer notifications if the backend has been away for a second and flood the user with tons of "timer removed", and "timer added" spam. Could be confusing to have that in the activity log, but the change is nice in general, thanks.

**Comment by FernetMenta** [comment]
When you send an event, it makes use of several globals like window manager, thread messages, etc. Estimating risk is not easy. Even backports to Isengard have broken things though risk was considered low. 
You just voted against this: https://github.com/xbmc/xbmc/pull/8270 which wasn't a fix either. In order to make decisions simple and avoid conflict I'd say ONLY fixes are allowed.

**Comment by Jalle19** [comment]
@da-anda sadly tvheadend is one of those :-)

**Comment by ksooo** [comment]
I voted against #8270 because it contained lots of changed code (compared to this PR) without any functional change. This PR is different - few code changes, but end-user relevant functional enhancements.

Currently, we're working *towards* first beta, but it is not yet released. In this phase smaller improvements are imo still okay. Once beta 1 is out, work mode should change and only fixes should be allowed.  A beta is supposed not be functionally perfect, but feature complete - otherwise it would be a RC. This is how other projects/products I know work and this makes sense for me.

But as I said, I can live with not having this PR in Jarvis. I suggest let @MartijnKaijser have the final word on this.

**Comment by FernetMenta** [comment]
> A beta is supposed not be functionally perfect, but feature complete

an app like kodi is never feature complete. and the more we move to RERO this is even less important. 

Note that for PVR related changes special care is required. You only tested this with a single backend (tvh), right? Did you test adding new timer on the backend and trigger update while playing video? I am sure there is a whole bunch of untested scenarios related to this change.

**Comment by ksooo** [comment]
If you put it this way I'm of course doomed.

But maybe it was just dumb to make a difference between functional and non functional changes and to vote against a non functional change PR while still in alpha phase. Maybe this was a mistake but exactlyc this made myself points for this discussion. :-(

Anyway, meanwhile my point of few is that there should be no early and late alpha phase wrt different rules for PR content. THIS would make things difficult to handle. Rules should change for different stages - not while in a single phase.

**Comment by ksooo** [comment]
* pointless

**Comment by FernetMenta** [comment]
I thought we are in beta already. Am I wrong?

**Comment by ksooo** [comment]
We released alpha 4 and working towards beta 1, which is due Nov 15. ;-)

**Comment by FernetMenta** [comment]
the log says BETA1, hence the current code is beta.

**Comment by ksooo** [comment]
which log?

**Comment by FernetMenta** [comment]
kodi.log
Starting Kodi (16.0-BETA1 Git:2015-11-10-7914216-dirty). Platform: OS X x86 64-bit

This means that we kind of release beta to the public Nov, 14th

**Comment by FernetMenta** [comment]
this is what @MartijnKaijser sent out to all
 
````
Current Alpha feature window runs to 20 Oktober
Alpha 4 release on 1 November
After we'll start the beta windows for fixes only.
```

**Comment by MartijnKaijser** [comment]
I have to agree with @fernetmenta here. It's certainly a nice feature however it's not even really used yet by other kodi parts. Having pvr start using this actively with variety of backends makes it a potential time bomb. Code base has improved massively over the past years but even 15 has proven that a tiny fix in pvr or skinning has unwanted side affects. As such caution is needed until we have a better grasp on it. 
Indeed it's already in beta as well with target at 14th.

**Comment by ksooo** [comment]
Okay, no problem with that - as I wrote earlier. All good. ;-)

File: xbmc/pvr/addons/PVRClients.h (hunk 1 above)
Comment (xhaggi) [inline xbmc/pvr/addons/PVRClients.h]: in that case we should add a new method `GetFriendlyName()` instead of passing a bool in. much more friendly :smile:

Comment (ksooo) [inline xbmc/pvr/addons/PVRClients.h]: No problem. Will do that.

Comment (ksooo) [inline xbmc/pvr/addons/PVRClients.h]: Done.

**Comment by xhaggi** [comment]
@ksooo thanks

**Comment by ksooo** [comment]
jenkins  build this please

**Comment by ksooo** [comment]
jenkins  build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**